common/djangoapps/util/views.py

```diff
@@ -218,7 +218,10 @@ def reset_course_deadlines(request):
         user = masquerade_user
     else:
         user = request.user
-    reset_self_paced_schedule(user, course_key)
+
+    missed_deadlines, missed_gated_content = dates_banner_should_display(course_key, request)
+    if missed_deadlines and not missed_gated_content:
```

Comment:
This has turned into a permissions check in some ways (do you have permission to reset your schedule), which makes me wonder/think that it should be pushed to a `has_perm` check, rather than a direct check here. OTOH, I'm not sure if we actually get much value out of it immediately, since there's no way it's going to be something we can read out of the database w/ bridgekeeper.